### PR TITLE
Make repo_name var to allow user specify different docker repo and increase Jenkins docker build timeout

### LINKED ISSUE #3358: [Bug]: Gradle check workflows are failing with docker throttling
### Describe the bug

As far as I can tell, all gradle check workflows in the opensearch-project/OpenSearch repository are failing with some version of the following error:

```
> Task :test:fixtures:gcs-fixture:composeBuild FAILED
Building gcs-fixture
Service 'gcs-fixture' failed to build: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

https://build.ci.opensearch.org/job/gradle-check/13164/ is the first instance I have found and it ran at 6:17:46 PM (UTC)

### To reproduce

Do anything to trigger a gradle check workflow (e.g. submit a PR)

### Expected behavior

_No response_

### Host / Environment

_No response_

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Make repo_name var to allow user specify different docker repo and increase Jenkins docker build timeout

### Issues Resolved
https://github.com/opensearch-project/opensearch-benchmark/issues/255

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
The docker build complete at the last second:
https://build.ci.opensearch.org/job/docker-build/3212/console

File: docker/ci/build-image-multi-arch.sh [hunk 1]
```diff
@@ -46,12 +47,15 @@ function cleanup_all() {
...
     File_Delete $DIR
 }
-while getopts ":hv:f:" arg; do
+while getopts ":hr:v:f:" arg; do
     case $arg in
         h)
             usage
             exit 1
             ;;
+        r)
```
Comment (zelinh) [inline docker/ci/build-image-multi-arch.sh:56]: Are we gonna input this `REPO_NAME` from the jenkins build script command in the parameter when running `docker_build`?

Comment (peterzhuamazon) [reply docker/ci/build-image-multi-arch.sh:56]: No, that one is using the scripts in docker/release dir, not in docker/ci dir like in this PR.

Comment (zelinh) [reply docker/ci/build-image-multi-arch.sh:56]: Just to make sure, with this change, we would need to add `REPO_NAME` here? https://github.com/opensearch-project/opensearch-build/blob/31991db16bd1d83bfc0e380a9c130994b9e40314/jenkins/docker/docker-build.jenkinsfile#L19-L24

Comment (peterzhuamazon) [reply docker/ci/build-image-multi-arch.sh:56]: I can tweak the description of this, and the defaultvalue.

Comment (peterzhuamazon) [reply docker/ci/build-image-multi-arch.sh:56]: Resolved.

**Review by zelinh — APPROVED** [review]
LGTM.

**Pull request merged**